src/lib/translate/simplifyNoopMoiveClips.js

```diff
@@ -0,0 +1,293 @@
+/**
+ *************************************************************************************
+ * Note this method modifies the input schema and then returns it
+ *************************************************************************************
+ *
+ * Translates "noop movie clips" (a clip with only shapes and noop tweens on those shapes) containers.
+ * This allows us to generate a segmented sprite sheet with more performant operations.
+ *
+ * The current implementation handles only movie clips with shapes as direct children and no action
+ * tweens.  It could be extended to support containers with only shapes as well as an additional type
+ * of child.
+ */
+import AnimateNode from '../parse/AnimateNode'
+import AnimateNodeReference from '../parse/AnimateNodeReference'
+
+function replaceMovieClipReferenceWithContainerInNativeObject (nativeObject, movieClip, container) {
+  const resolvedMovieClip = movieClip.node
+  const resolvedNativeObject = nativeObject.node
+
+  const nativeObjectData = resolvedNativeObject.data.object
+
+  for (const [ key, value ] of Object.entries(nativeObjectData)) {
+    if (value instanceof AnimateNode) {
+      let resolvedValue = value.node
+
+      // If this is an object we have a complex type that we need to unwind
+      if (resolvedValue.type === 'movie_clip') {
+        if (resolvedValue.id === resolvedMovieClip.id) {
+          // TODO consolidate this with the code below
+          const nodeReference = new AnimateNodeReference(
+            'TEMP', // ID will be finalized on the next line,
+            container.id,
+            { parsed: { [container.id]: container } } // Simulate target cache
+          )
+
+          nodeReference.finalizeId()
+          // TODO fix schema.references.push(nodeReference)
```

Comment:
What does this mean?

Reply:
Resolved - needed to pass schema through and was focusing on getting core functionality working.  Great catch.